src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionBuilderExtensions.cs

```diff
@@ -31,28 +31,9 @@ public static IApplicationBuilder UseHttpsRedirection(this IApplicationBuilder a
             }
 
             var options = app.ApplicationServices.GetRequiredService<IOptions<HttpsRedirectionOptions>>().Value;
+            var config = app.ApplicationServices.GetRequiredService<IConfiguration>();
 
-            // The tls port set in options will have priority over the one in configuration.
-            var httpsPort = options.HttpsPort;
-            if (httpsPort == null)
-            {
-                // Only read configuration if there is no httpsPort
-                var config = app.ApplicationServices.GetRequiredService<IConfiguration>();
-                var configHttpsPort = config["HTTPS_PORT"];
-                // If the string isn't empty, try to parse it.
-                if (!string.IsNullOrEmpty(configHttpsPort)
-                    && int.TryParse(configHttpsPort, out var intHttpsPort))
-                {
-                    httpsPort = intHttpsPort;
-                }
-            }
-
-            var rewriteOptions = new RewriteOptions();
-            rewriteOptions.AddRedirectToHttps(
-                options.RedirectStatusCode,
-                httpsPort);
-
-            app.UseRewriter(rewriteOptions);
+            app.UseMiddleware<HttpsRedirectionMiddleware>(app.ServerFeatures.Get<IServerAddressesFeature>(), config);
```

Comment:
You don't have to pass in the config here. If declared on the constructor it will just get injected.

Reply:
Too bad there's no way to populate IServerAddressesFeature for In-Proc ANCM. Right?

Reply:
I'll make an issue for this (to at least investigate whether we can populate it).

Reply:
https://github.com/aspnet/IISIntegration/issues/485

Reply:
Don't for get @javiercn 's comment about not needing to resolve IConfig here.

Reply:
whoopz